modules/addNavigation.js

```diff
@@ -0,0 +1,37 @@
+/**
+ * History enhancer that adds additional navigation methods — `transitionTo()`
+ * and `replaceWith()`. TODO: move to history module (?)
+ * @param {CreateHistory} createHistory - History-creating function
+ * @returns {CreateHistory}
+ */
+export default function addNavigation(createHistory) {
+  return (...args) => {
+    const history = createHistory(...args);
```

Comment:
Started writing a response and it got very long, so I moved it into the main thread: https://github.com/rackt/react-router/pull/1728#issuecomment-132860967